fobi_custom/plugins/form_handlers/collect_data/fobi_form_handlers.py

```diff
@@ -54,21 +54,27 @@ def run(self, form_entry, request, form, form_element_entries=None):
             plugin_data = element.plugin_data
             json_plugin_data = json.loads(plugin_data)
 
-            name = json_plugin_data['name']
-            label = json_plugin_data['label']
-            type = element.plugin_uid
-            position = element.position
-
-            saved_data = form.cleaned_data[name]
-
-            SurveyComponent.objects.create(
-                row=new_survey_row,
-                name=name,
-                label=label,
-                type=type,
-                position=position,
-                saved_data=saved_data
-            )
+            try:
+                name = json_plugin_data['name']
+
+                label = json_plugin_data['label']
+                type = element.plugin_uid
+                position = element.position
+
+                saved_data = form.cleaned_data[name]
+
+                SurveyComponent.objects.create(
+                    row=new_survey_row,
+                    name=name,
+                    label=label,
+                    type=type,
+                    position=position,
+                    saved_data=saved_data
+                )
+
+            # Help text elements have no name and should not be saved
+            except KeyError:
+                pass
```

Comment:
At what point do we expect this `KeyError` will get raised? On the assignment `name = json_plugin_data['name']`? It's best practice to keep `try/except` blocks scoped as tightly as possible to the blocks that you expect to raise errors -- in this case:

```python
try:
   name = json_plugin_data['name']
except KeyError:
   continue  # Skip to the next object in the enclosing loop
```

I'm also wondering if there's any more specific way that we can check that the field is for help text other than to see whether it raises a `KeyError` here. Seems like we're assuming that form fields will be missing the `name` attribute if and only if they're help text components, and I don't know how safe of an assumption that is. Does Fobi expose any identifying information about help text components that we could use to perform a more specific check? E.g. is the `plugin_uid` attribute passed into the form at all?

Reply:
I see your reasoning! The only plugin data that `content_text` pass into the form is a `text` key, but that's the only element that has it. I've switched to checking for that!